src/genie/libs/parser/sros/show_service_sap_using.py

```diff
@@ -0,0 +1,89 @@
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Any, Or, Optional
+import re
+
+# ======================================================
+# Schema for 'show service sap-using'
+# ======================================================
+
+class ShowServiceSapUsingSchema(MetaParser):
+    """schema for show service sap-using"""
+    schema = {
+        'total': int,
+        'sap': {
+            Any(): {
+                'service_id': int,
+                'ingress_qos': int,
+                'egress_qos': int,
+                'ingress_filter': str,
+                'egress_filter': str,
+                'admin_state': str,
+                'oper_state': str,
+            }
+        }
+    }
+
+class ShowServiceSapUsing(ShowServiceSapUsingSchema):
+    """ Parser for show service sap-using"""
+    cli_command = 'show service sap-using'
+
+    def cli(self, output=None):
+
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        result_dict = {}
+        # 2/1/6:10.*                      10         1     none    1     none   Up   Up
+        # 2/1/2:8.1                       181        1     none    1     none   Up   Down
+        # 2/1/2:3101.3101                 1121       3501  none    3501  none   Up   Down
+        # 2/1/6:*.*                    1121       1     none    1     none   Up   Up
+        # 2/1/6:3101.*                    1121       4503  none    4503  none   Up   Up
+        # lag-2:4000.*                    1122       1     none    1     none   Up   Up
+        # lag-1:400.1                    1221       1     none    1     none   Up   Up
+        p1 = re.compile(r'(?P<sap>(((\d+\/\d+\/\d+)|(lag-\d*))(:\d+|[*])*([.]\d+|[.*])*))'
```

Comment:
I admittedly don't know much about SROS. Will `(?P<sap>(((\d+\/\d+\/\d+)|(lag-\d*))(:\d+|[*])*([.]\d+|[.*])*))` cover all possibilities? Would a simpler `(?P<sap>\S+)` work better so we don't potentially have to go back and fix it should this have a different pattern?

Reply:
Hi, it will cover all standard ethernet type connections.
However, I prefer your simpler method as this should catch any differing format of sap as you suggest.

Many thanks for your feedback
Will amend.

Reply:
Hi, I need to add in support for some alternative hardware that I did not have access to.  Though SR-OS is generic, the alternative hardware will be missing one of the columns for this - so I am adding a secondary regex match and also making the variable optional in the schema.

Reply:
Sounds good. I don't see the changes yet but once they're in we can look at getting it merged